### PR TITLE
Pacifists can no longer cause harm using mech equipment

### PR DESCRIPTION
**Pull request by Irafas**
_Pacifists have learned that the missiles, bullets and lasers that mechs fire cause injury._

adds new variable to mech equipment  called pacifist_safe. A pacifist can only use this equipment when this is set to true. 

:cl: 
fix: Fixed pacifists from being able to fire mech weapons
/:cl:

Fixes issue #36276

### CODE REVIEW

**Comment by XDTM** [comment]
use TRUE and FALSE pls

**Comment by Irafas** [comment]
alright code has been fixed. Still new to programming in Byond and from the work I did for my previous PR I misinterpreted how booleans work.

**Comment by SpaceManiac** [comment]
It might be necessary to do checks for the drill and hydraulic clamp based on target type. Otherwise pacifism will prohibit loading/unloading cargo and mining rock.

**Comment by Bawhoppen** [comment]
That sounds alot like harming rocks to me.

**Comment by Irafas** [comment]
Already take care of. In mecha.dm I added some checks to see if the equipment is melee and targeting a human. It means no clamping of humans but ores can be clamped as much as you want.

Basically if the pacifist unsafe equipment is ranged it will always be impossible for a pacifist to fire it (as harm could occur from crossfire or AOE). If its melee equipment though it will check if the target is human.

**Comment by Irafas** [comment]
Well it actually can't melee hit anything living. So no clamping of Ian either

**Comment by SpaceManiac** [comment]
As long as rock can be drilled and ore boxes/crates/closets can be clamped, it should be fine.
Actually drilling closets may be a problem. I think, but I'm not sure, that drilling a closet containing a person hurts the person.

**Comment by Irafas** [comment]
Would certainly be a funny way to get around not harming people.

Still it's like nearly four in the morning for me and the only reason I'm here is because my phone beeped so loud it woke me up. I'll give the locker scenario a test when I wake up.

**Comment by 81Denton** [comment]
@Irafas Drilling a locker with a person inside resulted in a little accident 

>The APLU "Firefighter" starts to drill the electrical supplies locker.
*---------*
This is Flip Zadovsky!
He is limp and unresponsive; there are no signs of life and his soul has departed...
His right arm is missing!
He has severe bruising!
He has severe burns!
He is bleeding!
*---------*

Haha oops

**Comment by PKPenguin321** [comment]
Drills calling ex_act on the contents of shit they drill can be blamed for that

**Comment by Irafas** [comment]
Alright so now pacifists can no longer drill or clamp opened lockers and crates. This applies even if the container doesn't have a living being in it, to avoid the player having the ability to know if someone is in something without even checking. This does cause the unfortunate effect that loading crates full of things into the cargo bay of a mech is now impossible. So I may just change it so that it checks if a living being is in the container.

**Comment by Irafas** [comment]
Changed it so that the game checks if the container has a living being in it. If it does the pacifist can not drill, clamp, or melee it. This does mean that pacifists essentially have mind powers that let them know if someone is in a container without having to open it. Still this means that pacifists can now load crates with cargo, unlike in my previous commit.

File: code/game/mecha/mecha.dm [hunk 1]
```diff
@@ -440,11 +440,24 @@
...
+				to_chat(user, "<span class='warning'>You don't want to harm other living beings!</span>")
+				return
 			if(selected.action(target,params))
 				selected.start_cooldown()
 	else if(selected && selected.is_melee())
+		if(isliving(target) && !selected.pacifist_safe && L.has_trait(TRAIT_PACIFISM))
+			to_chat(user, "<span class='warning'>You don't want to harm other living beings!</span>")
+			return
+		if(istype(C) && L.has_trait(TRAIT_PACIFISM) && !selected.pacifist_safe)
+			for(var/mob/living/M in C.contents)
```
Comment (duncathan) [inline code/game/mecha/mecha.dm]: tiny nitpick, you can just do `in C` instead of `in C.contents`

**Review by duncathan — CHANGES_REQUESTED** [review]

**Comment by SpaceManiac** [comment]
The clamp could still be safe to use on containers with livings in them. You can put humans in lockers and carry them around non-harmfully in the cargo compartment.

File: code/game/mecha/mecha.dm (hunk 1 above)
Comment (Irafas) [reply code/game/mecha/mecha.dm]: alright thanks. Made the change. Also do you know of any better ways to check the contents of a container? Having it do a for loop felt a bit hacky.

**Comment by Irafas** [comment]
Alright the clamp can now be used to pick up containers that have people in them.

**Pull request merged**